src/isomorphic/ReactDebugTool.js

```diff
@@ -49,6 +101,95 @@ var ReactDebugTool = {
       }
     }
   },
+  beginProfiling() {
+    if (__DEV__) {
+      if (isProfiling) {
+        return;
+      }
+
+      isProfiling = true;
+      flushHistory.length = 0;
+      resetMeasurements();
+    }
+  },
+  endProfiling() {
+    if (__DEV__) {
+      if (!isProfiling) {
+        return;
+      }
+
+      isProfiling = false;
+      resetMeasurements();
+    }
+  },
+  getFlushHistory() {
+    if (__DEV__) {
+      return flushHistory;
+    }
+  },
+  onBeginFlush() {
```

Comment:
Can you explain a bit on why the flush batch is important? Does it matter if the implementation batches or just does the work whenever?